app/src/main/java/net/gsantner/opoc/frontend/GsSearchOrCustomTextDialog.java

```diff
@@ -130,14 +145,29 @@ public static class DialogOptions {
         public int clearInputIcon = android.R.drawable.ic_menu_close_clear_cancel;
         @StyleRes
         public int dialogStyle = 0;
+
+        /**
+         * Initial state of the dialog. Will be updated when the dialog is dismissed.
+         */
+        public final DialogState state = new DialogState();
+
+        /**
+         * Reference to the dialog. Will be updated when the dialog is created.
+         */
+        public WeakReference<AlertDialog> dialog = null;
     }
 
     public static class DialogState {
         public int listPosition = -1;
-        public String defaultText = "";
-        public String searchQuery = "";
-        public Parcelable instanceState;
-        public Set<Integer> disabledLevels = new HashSet<>(); // For HeadlineDialog
+        public String searchText = "";
```

Comment:
We can cleanly set the dialog state in the beginning and observe it from callbacks etc